c-api/compat-5.3.c

```diff
@@ -336,22 +392,157 @@ COMPAT53_API void luaL_traceback (lua_State *L, lua_State *L1,
 
 
 COMPAT53_API int luaL_fileresult (lua_State *L, int stat, const char *fname) {
+  const char* s = NULL;
   int en = errno;  /* calls to Lua API may change this value */
   if (stat) {
     lua_pushboolean(L, 1);
     return 1;
   }
   else {
     lua_pushnil(L);
+#if defined(__GLIBC__) || defined(_POSIX_VERSION) || defined(__APPLE__) || (defined(__GNUC__) && (__GNUC__ < 6) && defined (__MINGW32__))
+    // use strerror_r here, because it's available on these specific platforms
+    char buf[512];
+    strerror_r(en, buf, 1024);
+    s = buf;
+#elif defined(_MSC_VER)
+    // for MSVC and others, use strerror_s since it's provided by default by the libraries
+    char buf[512];
+    strerror_s(buf, 1024, en);
+    s = buf;
+#else
+    s = strerror(en);
```

Comment:
maybe add at least a comment that this is unsafe since it's not thread-safe, or implement your own version (see comment above)